requirements-dev.txt

```diff
@@ -0,0 +1,4 @@
+pytest>=2.7.2
+pre-commit>=3.5.0
+mypy
+ruff
```

Comment:
do we really need that? as far as I understood the tools are already installed with the extension

Reply:
Linting in CI still requires ruff to be installed via requirements.

Reply:
since we run the linter also in CI via pre-commit it should not be needed. pre-commit is installing all the needed tools on the fly.

Reply:
done